
2018/02/24 10:16:43 [error] 44026#0: *1 lua entry thread aborted: runtime error: content_by_lua(nginx.conf:35):2: module 'mobdebug' not found:
	no field package.preload['mobdebug']
	no file '../../debugger/lualibs/mobdebug/mobdebug.lua'
	no file '../../debugger/lualibs/mobdebug.lua'
	no file 'mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug/mobdebug.lua'
	no file '/usr/local/openresty/site/lualib/mobdebug.ljbc'
	no file '/usr/local/openresty/site/lualib/mobdebug/init.ljbc'
	no file '/usr/local/openresty/lualib/mobdebug.ljbc'
	no file '/usr/local/openresty/lualib/mobdebug/init.ljbc'
	no file '/usr/local/openresty/site/lualib/mobdebug.lua'
	no file '/usr/local/openresty/site/lualib/mobdebug/init.lua'
	no file '/usr/local/openresty/lualib/mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug/init.lua'
	no file './mobdebug.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/mobdebug.lua'
	no file '/usr/local/share/lua/5.1/mobdebug.lua'
	no file '/usr/local/share/lua/5.1/mobdebug/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/mobdebug.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/mobdebug/init.lua'
	no file '../../debugger/clibs/mobdebug.dylib'
	no file '/usr/local/openresty/lualib/mobdebug.dylib'
	no file '/usr/local/openresty/lualib/mobdebug/mobdebug.dylib'
	no file '/usr/local/openresty/site/lualib/mobdebug.so'
	no file '/usr/local/openresty/lualib/mobdebug.so'
	no file './mobdebug.so'
	no file '/usr/local/lib/lua/5.1/mobdebug.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/mobdebug.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
coroutine 0:
	[C]: in function 'require'
	content_by_lua(nginx.conf:35):2: in function <content_by_lua(nginx.conf:35):1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:20:00 [error] 44246#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:21:50 [error] 44246#0: *3 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:26:30 [error] 44792#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:27:02 [error] 44792#0: *4 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:27:20 [error] 44792#0: *3 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:27:20 [alert] 44792#0: *4 setsockopt(TCP_NODELAY) failed (22: Invalid argument) while keepalive, client: 127.0.0.1, server: 0.0.0.0:8010
2018/02/24 10:36:31 [error] 45370#0: *1 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:2: module 'mobdebug' not found:
	no field package.preload['mobdebug']
	no file '../debugger/mobdebug/mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug/mobdebug.lua'
	no file '/usr/local/openresty/site/lualib/mobdebug.ljbc'
	no file '/usr/local/openresty/site/lualib/mobdebug/init.ljbc'
	no file '/usr/local/openresty/lualib/mobdebug.ljbc'
	no file '/usr/local/openresty/lualib/mobdebug/init.ljbc'
	no file '/usr/local/openresty/site/lualib/mobdebug.lua'
	no file '/usr/local/openresty/site/lualib/mobdebug/init.lua'
	no file '/usr/local/openresty/lualib/mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug/init.lua'
	no file './mobdebug.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/mobdebug.lua'
	no file '/usr/local/share/lua/5.1/mobdebug.lua'
	no file '/usr/local/share/lua/5.1/mobdebug/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/mobdebug.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/mobdebug/init.lua'
	no file '/usr/local/openresty/site/lualib/mobdebug.so'
	no file '/usr/local/openresty/lualib/mobdebug.so'
	no file './mobdebug.so'
	no file '/usr/local/lib/lua/5.1/mobdebug.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/mobdebug.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
coroutine 0:
	[C]: in function 'require'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:2: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:36:32 [error] 45370#0: *2 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:2: module 'mobdebug' not found:
	no field package.preload['mobdebug']
	no file '../debugger/mobdebug/mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug/mobdebug.lua'
	no file '/usr/local/openresty/site/lualib/mobdebug.ljbc'
	no file '/usr/local/openresty/site/lualib/mobdebug/init.ljbc'
	no file '/usr/local/openresty/lualib/mobdebug.ljbc'
	no file '/usr/local/openresty/lualib/mobdebug/init.ljbc'
	no file '/usr/local/openresty/site/lualib/mobdebug.lua'
	no file '/usr/local/openresty/site/lualib/mobdebug/init.lua'
	no file '/usr/local/openresty/lualib/mobdebug.lua'
	no file '/usr/local/openresty/lualib/mobdebug/init.lua'
	no file './mobdebug.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/mobdebug.lua'
	no file '/usr/local/share/lua/5.1/mobdebug.lua'
	no file '/usr/local/share/lua/5.1/mobdebug/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/mobdebug.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/mobdebug/init.lua'
	no file '/usr/local/openresty/site/lualib/mobdebug.so'
	no file '/usr/local/openresty/lualib/mobdebug.so'
	no file './mobdebug.so'
	no file '/usr/local/lib/lua/5.1/mobdebug.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/mobdebug.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
coroutine 0:
	[C]: in function 'require'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:2: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:37:38 [error] 45482#0: *1 lua entry thread aborted: runtime error: ../debugger/lualibs/mobdebug/mobdebug.lua:102: loop or previous error loading module 'socket'
stack traceback:
coroutine 0:
	[C]: in function 'require'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:2: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:37:50 [error] 45482#0: *2 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:2: loop or previous error loading module 'mobdebug'
stack traceback:
coroutine 0:
	[C]: in function 'require'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:2: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:39:44 [error] 45660#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:41:54 [error] 45660#0: *3 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:42:42 [error] 45660#0: *3 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:46:09 [error] 46194#0: *12 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:53:52 [error] 46194#0: *14 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:53:56 [error] 46194#0: *14 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:54:19 [error] 46194#0: *14 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:55:04 [error] 46194#0: *14 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:55:10 [error] 46194#0: *14 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:56:08 [error] 46684#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:56:36 [error] 46725#0: *3 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 10:58:47 [emerg] 46861#0: directive "lua_package_path" is not terminated by ";" in /Users/ryuu/git/tutorials/openresty/redis-bitmap/conf/nginx.conf:11
2018/02/24 10:59:15 [emerg] 46912#0: bind() to 0.0.0.0:8010 failed (48: Address already in use)
2018/02/24 10:59:15 [emerg] 46912#0: bind() to 0.0.0.0:8010 failed (48: Address already in use)
2018/02/24 10:59:15 [emerg] 46912#0: bind() to 0.0.0.0:8010 failed (48: Address already in use)
2018/02/24 10:59:15 [emerg] 46912#0: bind() to 0.0.0.0:8010 failed (48: Address already in use)
2018/02/24 10:59:15 [emerg] 46912#0: bind() to 0.0.0.0:8010 failed (48: Address already in use)
2018/02/24 11:00:11 [error] 46958#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:00:26 [error] 46958#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:04:07 [error] 46958#0: *4 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:04:21 [error] 46958#0: *4 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:3: Debugger connection closed
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:04:31 [error] 46958#0: *6 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:04:38 [error] 46958#0: *6 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:06:42 [error] 47355#0: *9 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:06:49 [error] 47355#0: *9 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:18:11 [error] 47355#0: *9 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:19:23 [error] 47355#0: *9 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:19:58 [error] 47355#0: *9 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:21:53 [error] 47355#0: *15 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:24:11 [error] 48245#0: *17 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:24:32 [error] 48245#0: *17 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:25:50 [error] 48245#0: *17 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:27:05 [error] 48245#0: *21 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:27:53 [error] 48245#0: *21 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:33:55 [error] 48245#0: *24 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:34:14 [error] 48245#0: *24 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:34:33 [error] 48245#0: *24 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:38:32 [error] 48245#0: *28 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:40:11 [error] 48245#0: *30 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:40:31 [error] 48245#0: *30 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:40:52 [error] 48245#0: *30 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:41:29 [error] 48245#0: *30 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:42:11 [error] 49148#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:42:35 [error] 49148#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:43:26 [error] 49222#0: *4 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:43:48 [error] 49222#0: *4 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:44:16 [error] 49222#0: *4 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:44:37 [error] 49222#0: *4 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:45:13 [error] 49292#0: *10 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:45:31 [error] 49292#0: *10 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:45:37 [error] 49292#0: *10 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:45:47 [error] 49292#0: *10 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:45:52 [error] 49292#0: *10 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:46:42 [error] 49398#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 11:47:11 [error] 49398#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:17:05 [alert] 51367#0: lua_code_cache is off; this will hurt performance in /Users/ryuu/git/tutorials/openresty/redis-bitmap/conf/nginx.conf:12
2018/02/24 13:17:49 [error] 51369#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:18:11 [error] 51369#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:18:35 [error] 51369#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:18:47 [error] 51369#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:34:24 [error] 51369#0: *6 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:50:25 [error] 51369#0: *8 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:52:18 [error] 51369#0: *10 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:52:21 [error] 51369#0: *10 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:252: bad argument #1 to 'rawget' (table expected, got nil)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:252: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:489: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:52:21 [error] 51369#0: *12 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:52:22 [error] 51369#0: *12 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:252: bad argument #1 to 'rawget' (table expected, got nil)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:252: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:489: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:52:22 [error] 51369#0: *14 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:52:23 [error] 51369#0: *14 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:252: bad argument #1 to 'rawget' (table expected, got nil)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:252: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:489: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:53:50 [error] 51369#0: *16 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:54:20 [error] 51369#0: *16 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:54:45 [error] 51369#0: *16 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:55:43 [error] 51369#0: *20 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:56:54 [error] 51369#0: *22 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 13:59:40 [error] 51369#0: *24 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:06:11 [error] 51369#0: *26 failed to load external Lua file "/Users/ryuu/git/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua": ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:490: function arguments expected near '.', client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:06:12 [error] 51369#0: *27 failed to load external Lua file "/Users/ryuu/git/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua": ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:490: function arguments expected near '.', client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:08:26 [error] 51369#0: *28 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:11:04 [error] 51369#0: *28 lua tcp socket read timed out, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:11:13 [error] 51369#0: *28 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:16:42 [error] 51369#0: *31 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:36:35 [error] 51369#0: *31 lua tcp socket read timed out, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:36:47 [error] 51369#0: *31 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:47:06 [error] 51369#0: *34 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:47:08 [error] 51369#0: *34 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: attempt to call field 'GETCOUNTEXP' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:47:08 [error] 51369#0: *36 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:47:09 [error] 51369#0: *36 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: attempt to call field 'GETCOUNTEXP' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:47:09 [error] 51369#0: *38 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:47:10 [error] 51369#0: *38 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: attempt to call field 'GETCOUNTEXP' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:47:25 [error] 51369#0: *40 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:02 [error] 57465#0: *43 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:03 [error] 57465#0: *43 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: attempt to call field 'GETCOUNTEXP' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:20 [error] 57465#0: *45 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:23 [error] 57465#0: *45 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.GETCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:24 [error] 57465#0: *47 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:25 [error] 57465#0: *47 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.GETCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:26 [error] 57465#0: *49 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:27 [error] 57465#0: *49 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.GETCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:32 [error] 57465#0: *51 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:48:35 [error] 57465#0: *51 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.GETCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:49:08 [error] 57465#0: *53 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:49:09 [error] 57465#0: *53 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.GETCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:49:13 [error] 57465#0: *55 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:49:14 [error] 57465#0: *55 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.GETCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:49:52 [error] 57465#0: *57 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:49:54 [error] 57465#0: *57 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.BITCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:49:57 [error] 57465#0: *59 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:49:58 [error] 57465#0: *59 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.BITCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:50:50 [error] 57465#0: *61 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:50:51 [error] 57465#0: *61 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.BITCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:52:34 [error] 57465#0: *63 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:52:41 [error] 57465#0: *63 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.BITCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:52:44 [error] 57465#0: *65 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:52:47 [error] 57465#0: *65 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.BITCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:53:00 [error] 57465#0: *67 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:53:01 [error] 57465#0: *67 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: bad argument #1 to 'rawget' (table expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'rawget'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:253: in function '_do_cmd'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:335: in function 'R.BITCOUNTEXP'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:494: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:53:44 [error] 57465#0: *69 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:57:31 [error] 57465#0: *71 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 14:57:49 [error] 57465#0: *71 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:11 [error] 57465#0: *74 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:14 [error] 57465#0: *74 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:15 [error] 57465#0: *76 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:16 [error] 57465#0: *76 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:17 [error] 57465#0: *78 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:21 [error] 57465#0: *78 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:42 [error] 57465#0: *80 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:44 [error] 57465#0: *80 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:44 [error] 57465#0: *82 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:45 [error] 57465#0: *82 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:45 [error] 57465#0: *84 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:00:45 [error] 57465#0: *84 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:01:01 [error] 57465#0: *86 failed to load external Lua file "/Users/ryuu/git/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua": ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:495: function arguments expected near 'local', client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:01:45 [error] 57465#0: *87 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:01:48 [error] 57465#0: *87 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:01:48 [error] 57465#0: *89 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:01:49 [error] 57465#0: *89 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:01:49 [error] 57465#0: *91 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:01:49 [error] 57465#0: *91 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:493: attempt to call method 'R' (a table value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:03:06 [error] 57465#0: *93 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:04:15 [error] 57465#0: *95 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:06:18 [error] 57465#0: *95 lua tcp socket read timed out, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:09:44 [error] 57465#0: *97 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:11:38 [error] 57465#0: *99 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:25:33 [error] 57465#0: *101 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:65: attempt to call global '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:78: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:26:37 [alert] 59838#0: lua_code_cache is off; this will hurt performance in /Users/ryuu/git/tutorials/openresty/redis-bitmap/conf/nginx.conf:12
2018/02/24 15:27:37 [alert] 59937#0: lua_code_cache is off; this will hurt performance in /Users/ryuu/git/tutorials/openresty/redis-bitmap/conf/nginx.conf:12
2018/02/24 15:27:52 [error] 59939#0: *1 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:65: attempt to call global '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:78: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:32:08 [error] 59939#0: *2 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:51: attempt to call local '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:82: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:32:34 [error] 59939#0: *3 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:51: attempt to call local '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:82: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:32:35 [error] 59939#0: *4 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:51: attempt to call local '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:82: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:32:36 [error] 59939#0: *5 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:51: attempt to call local '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:82: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:33:59 [error] 59939#0: *6 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:51: attempt to call local '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:82: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:34:06 [error] 59939#0: *7 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:51: attempt to call local '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:82: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:34:09 [error] 59939#0: *8 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:51: attempt to call local '_do_cmd' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function 'set_time'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:82: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:54:59 [error] 59939#0: *9 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:54:59 [error] 59939#0: *9 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:74: bad argument #2 to 'connect' (number expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'connect'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:74: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:55:00 [error] 59939#0: *10 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:55:00 [error] 59939#0: *10 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:74: bad argument #2 to 'connect' (number expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'connect'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:74: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:56:22 [error] 59939#0: *11 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:56:22 [error] 59939#0: *11 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:74: bad argument #2 to 'connect' (number expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'connect'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:74: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:56:36 [error] 59939#0: *12 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:56:36 [error] 59939#0: *12 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:74: bad argument #2 to 'connect' (number expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'connect'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:74: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:57:40 [error] 59939#0: *13 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:58:33 [alert] 62094#0: lua_code_cache is off; this will hurt performance in /Users/ryuu/git/tutorials/openresty/redis-bitmap/conf/nginx.conf:12
2018/02/24 15:58:37 [error] 62096#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:58:38 [error] 62096#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:58:52 [error] 62096#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:59:02 [error] 62096#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:59:03 [error] 62096#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:59:04 [error] 62096#0: *1 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 15:59:53 [error] 62096#0: *2 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:00:34 [error] 62096#0: *3 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:00:34 [error] 62096#0: *3 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:77: bad argument #2 to 'connect' (number expected, got string)
stack traceback:
coroutine 0:
	[C]: in function 'connect'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:77: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:09:15 [alert] 62932#0: lua_code_cache is off; this will hurt performance in /Users/ryuu/git/tutorials/openresty/redis-bitmap/conf/nginx.conf:12
2018/02/24 16:09:27 [error] 62935#0: *1 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:5: module 'redis-extra' not found:
	no field package.preload['redis-extra']
	no file '../debugger/lualibs/redis-extra/redis-extra.lua'
	no file '../../debugger/lualibs/redis-extra.lua'
	no file 'redis-extra.lua'
	no file '/usr/local/openresty/site/lualib/redis-extra.ljbc'
	no file '/usr/local/openresty/site/lualib/redis-extra/init.ljbc'
	no file '/usr/local/openresty/lualib/redis-extra.ljbc'
	no file '/usr/local/openresty/lualib/redis-extra/init.ljbc'
	no file '/usr/local/openresty/site/lualib/redis-extra.lua'
	no file '/usr/local/openresty/site/lualib/redis-extra/init.lua'
	no file '/usr/local/openresty/lualib/redis-extra.lua'
	no file '/usr/local/openresty/lualib/redis-extra/init.lua'
	no file './redis-extra.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/redis-extra.lua'
	no file '/usr/local/share/lua/5.1/redis-extra.lua'
	no file '/usr/local/share/lua/5.1/redis-extra/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/redis-extra.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/redis-extra/init.lua'
	no file '../debugger/clibs/redis-extra.dylib'
	no file '/usr/local/openresty/lualib/redis-extra.dylib'
	no file '/usr/local/openresty/lualib/redis-extra/redis-extra.dylib'
	no file '/usr/local/openresty/site/lualib/redis-extra.so'
	no file '/usr/local/openresty/lualib/redis-extra.so'
	no file './redis-extra.so'
	no file '/usr/local/lib/lua/5.1/redis-extra.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/redis-extra.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
coroutine 0:
	[C]: in function 'require'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:5: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:09:41 [error] 62935#0: *2 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:5: module 'redis-extra' not found:
	no field package.preload['redis-extra']
	no file '../debugger/lualibs/redis-extra/redis-extra.lua'
	no file '../../debugger/lualibs/redis-extra.lua'
	no file 'redis-extra.lua'
	no file '/usr/local/openresty/site/lualib/redis-extra.ljbc'
	no file '/usr/local/openresty/site/lualib/redis-extra/init.ljbc'
	no file '/usr/local/openresty/lualib/redis-extra.ljbc'
	no file '/usr/local/openresty/lualib/redis-extra/init.ljbc'
	no file '/usr/local/openresty/site/lualib/redis-extra.lua'
	no file '/usr/local/openresty/site/lualib/redis-extra/init.lua'
	no file '/usr/local/openresty/lualib/redis-extra.lua'
	no file '/usr/local/openresty/lualib/redis-extra/init.lua'
	no file './redis-extra.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/redis-extra.lua'
	no file '/usr/local/share/lua/5.1/redis-extra.lua'
	no file '/usr/local/share/lua/5.1/redis-extra/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/redis-extra.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/redis-extra/init.lua'
	no file '../debugger/clibs/redis-extra.dylib'
	no file '/usr/local/openresty/lualib/redis-extra.dylib'
	no file '/usr/local/openresty/lualib/redis-extra/redis-extra.dylib'
	no file '/usr/local/openresty/site/lualib/redis-extra.so'
	no file '/usr/local/openresty/lualib/redis-extra.so'
	no file './redis-extra.so'
	no file '/usr/local/lib/lua/5.1/redis-extra.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/redis-extra.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
coroutine 0:
	[C]: in function 'require'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:5: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:09:58 [error] 62935#0: *3 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:5: module 'redis-extra' not found:
	no field package.preload['redis-extra']
	no file '../debugger/lualibs/redis-extra/redis-extra.lua'
	no file '../../debugger/lualibs/redis-extra.lua'
	no file 'redis-extra.lua'
	no file '/usr/local/openresty/site/lualib/redis-extra.ljbc'
	no file '/usr/local/openresty/site/lualib/redis-extra/init.ljbc'
	no file '/usr/local/openresty/lualib/redis-extra.ljbc'
	no file '/usr/local/openresty/lualib/redis-extra/init.ljbc'
	no file '/usr/local/openresty/site/lualib/redis-extra.lua'
	no file '/usr/local/openresty/site/lualib/redis-extra/init.lua'
	no file '/usr/local/openresty/lualib/redis-extra.lua'
	no file '/usr/local/openresty/lualib/redis-extra/init.lua'
	no file './redis-extra.lua'
	no file '/usr/local/openresty/luajit/share/luajit-2.1.0-beta3/redis-extra.lua'
	no file '/usr/local/share/lua/5.1/redis-extra.lua'
	no file '/usr/local/share/lua/5.1/redis-extra/init.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/redis-extra.lua'
	no file '/usr/local/openresty/luajit/share/lua/5.1/redis-extra/init.lua'
	no file '../debugger/clibs/redis-extra.dylib'
	no file '/usr/local/openresty/lualib/redis-extra.dylib'
	no file '/usr/local/openresty/lualib/redis-extra/redis-extra.dylib'
	no file '/usr/local/openresty/site/lualib/redis-extra.so'
	no file '/usr/local/openresty/lualib/redis-extra.so'
	no file './redis-extra.so'
	no file '/usr/local/lib/lua/5.1/redis-extra.so'
	no file '/usr/local/openresty/luajit/lib/lua/5.1/redis-extra.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
coroutine 0:
	[C]: in function 'require'
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:5: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:11:11 [alert] 63117#0: lua_code_cache is off; this will hurt performance in /Users/ryuu/git/tutorials/openresty/redis-bitmap/conf/nginx.conf:12
2018/02/24 16:11:16 [error] 63119#0: *1 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:7: attempt to index local 'redis' (a boolean value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:11:30 [error] 63119#0: *2 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:7: attempt to index local 'redis' (a boolean value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:12:46 [error] 63119#0: *3 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:12:53 [error] 63119#0: *3 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:44: attempt to index global '_M' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:12:53 [error] 63119#0: *5 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:12:55 [error] 63119#0: *5 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:44: attempt to index global '_M' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:12:55 [error] 63119#0: *7 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:12:57 [error] 63119#0: *7 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:44: attempt to index global '_M' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:13:54 [error] 63119#0: *9 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:13:58 [error] 63119#0: *9 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:44: attempt to index global '_M' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:13:58 [error] 63119#0: *11 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:13:59 [error] 63119#0: *11 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:44: attempt to index global '_M' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:13:59 [error] 63119#0: *13 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:13:59 [error] 63119#0: *13 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:44: attempt to index global '_M' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:10 [error] 63119#0: *15 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:20 [error] 63294#0: *17 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:22 [error] 63294#0: *17 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:44: attempt to index global '_M' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:22 [error] 63294#0: *19 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:23 [error] 63294#0: *19 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:44: attempt to index global '_M' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:36 [error] 63294#0: *21 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:39 [error] 63294#0: *21 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:46: attempt to call local 'v' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:39 [error] 63294#0: *23 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:41 [error] 63294#0: *23 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:46: attempt to call local 'v' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:41 [error] 63294#0: *25 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:42 [error] 63294#0: *25 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:46: attempt to call local 'v' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:50 [error] 63294#0: *27 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:14:51 [error] 63294#0: *27 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:46: attempt to call local 'v' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:15:16 [error] 63294#0: *29 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:15:16 [error] 63294#0: *29 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:46: attempt to call local 'v' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:15:17 [error] 63294#0: *31 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:15:17 [error] 63294#0: *31 lua entry thread aborted: runtime error: ...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:46: attempt to call local 'v' (a nil value)
stack traceback:
coroutine 0:
	...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua: in function <...it/tutorials/openresty/redis-bitmap/lua/redis-bitmap.lua:1>, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:15:30 [error] 63294#0: *33 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:16:09 [error] 63294#0: *35 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:16:16 [error] 63294#0: *35 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:16:46 [error] 63294#0: *35 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:17:27 [error] 63294#0: *35 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:18:04 [error] 63294#0: *35 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:18:52 [error] 63294#0: *41 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:19:02 [error] 63294#0: *41 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:43:32 [error] 63294#0: *44 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:44:03 [error] 63294#0: *44 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:44:30 [error] 63294#0: *44 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:44:31 [error] 63294#0: *44 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:45:03 [error] 63294#0: *44 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:45:04 [error] 63294#0: *44 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:45:05 [error] 63294#0: *44 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:46:11 [error] 63294#0: *52 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:46:43 [error] 63294#0: *54 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:47:16 [error] 63294#0: *56 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:47:29 [error] 63294#0: *56 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:47:38 [error] 63294#0: *56 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:48:04 [error] 63294#0: *56 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:49:32 [error] 63294#0: *61 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:49:34 [error] 63294#0: *61 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
2018/02/24 16:49:34 [error] 63294#0: *61 attempt to send data on a closed socket: u:0000000000000000, c:0000000000000000, ft:0 eof:0, client: 127.0.0.1, server: , request: "GET /redis HTTP/1.1", host: "localhost:8010"
